core/types/request.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+var (
+	ErrRequestTypeNotSupported = errors.New("request type not supported")
+	errShortTypedRequest       = errors.New("typed request too short")
+)
+
+// Request types.
+const (
+	DepositRequestType = 0x00
+)
+
+// Request is an EIP-7685 request object. It represents execution layer
+// triggered messages bound for the consesus layer.
+type Request struct {
+	inner RequestData
+}
+
+// Type returns the EIP-7685 type of the request.
+func (r *Request) Type() byte {
+	return r.inner.requestType()
+}
+
+// NewRequest creates a new request.
+func NewRequest(inner RequestData) *Request {
+	req := new(Request)
+	req.inner = inner.copy()
+	return req
+}
+
+// Requests implements DerivableList for requests.
+type Requests []*Request
+
+// Len returns the length of s.
+func (s Requests) Len() int { return len(s) }
+
+// EncodeIndex encodes the i'th request to s.
+func (s Requests) EncodeIndex(i int, w *bytes.Buffer) {
+	s[i].encode(w)
+}
+
+// Retrieve deposits from a requests list.
+func (s Requests) Deposits() Deposits {
```

Comment:
This is only used in the t8ntool, while the normal execution in block.Deposits does the check differently, I think we should harmonize the two and do both the type check and the cast check

Reply:
done